frontend/components/ReservationModal.js

```diff
@@ -0,0 +1,167 @@
+import { useState } from "react";
+import Modal from "@mui/material/Modal";
+import Paper from "@mui/material/Paper";
+import Typography from "@mui/material/Typography";
+import TextField from "@mui/material/TextField";
+import Button from "@mui/material/Button";
+
+export default function ReservationModal({
+  open,
+  onClose,
+  onSubmit,
+  mealId,
+  available_spots,
+}) {
+  const [form, setForm] = useState({
+    number_of_guests: 1,
+    contact_name: "",
+    contact_email: "",
+    contact_phonenumber: "",
+  });
+
+  const [phoneError, setPhoneError] = useState("");
+  const [guestsError, setGuestsError] = useState("");
+  const [emailError, setEmailError] = useState("");
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (
+      validatePhoneNumber(form.contact_phonenumber) &&
+      validateGuests(form.number_of_guests) &&
+      validateEmail(form.contact_email)
+    ) {
+      onSubmit({ ...form, meal_id: mealId });
+    }
+  };
+
+  const validatePhoneNumber = (phoneNumber) => {
+    const phoneRegex = /^\+?[0-9]{1,3}?[- ]?[0-9]{6,14}$/;
```

Comment:
maybe it would be nice to explain to the user why the phone number entered was invalid